ics/src/main/java/com/example/demo/controllers/ImagesController.java

```diff
@@ -37,25 +37,27 @@ public List<Image> listImages() {
     @GetMapping
     @RequestMapping("/{id}")
     public ResponseEntity<RecognitionResponseBody> getImage(@PathVariable Long id) {
-        if(!imageRepository.existsById(id)) return  ResponseEntity.notFound().build();
+        if (!imageRepository.existsById(id)) return ResponseEntity.notFound().build();
 
         return RecognitionResponseBody.imageToResponce(
                 imageRepository.getReferenceById(id)
                 , HttpStatus.OK);
     }
 
     @PostMapping
-    public ResponseEntity<RecognitionResponseBody> classifyImage(@RequestBody RecognitionRequestBody body) {
+    public ResponseEntity<RecognitionResponseBody> classifyImage(@RequestBody RecognitionRequestBody body
```

Comment:
I see that you have TODOs for validating the input data. Spring can help here. Please take a look at these resources:
Java bean validation: https://www.baeldung.com/javax-validation
`@Valid`  and `@Validated` annotations: https://www.baeldung.com/spring-valid-vs-validated

Reply:
adjusted